docs/DeveloperGuide.md

```diff
@@ -141,12 +141,57 @@ Classes used by multiple components are in the `seedu.clinic.commons` package.
 
 This section describes some noteworthy details on how certain features are implemented.
 
-{To be updated with individual enhancement}
-
-### \[Proposed\] Data archiving
-
-_{Explain here how the data archiving feature will be implemented}_
-
+### Find feature
+
+#### What Find feature does
+The find feature allows users to find all relevant suppliers or warehouses by their names, and/or by their remarks
+and/or by the products sold/stored. Users are able to search for relevant suppliers or warehouses using either only one
+of these criterion or a combination of these criteria. Note that users are only able to search for either suppliers or
```

Comment:
I feel like this second sentence is a bit of repeat with the and/or in the first sentence, maybe we can just remove the and/or in the first sentence?